### PR TITLE
feat: Text scrolling: allow to disable and reset on pause

### PR DESCRIPTION
**Pull request by luisbocanegra**
Closes: https://github.com/ccatterina/plasmusic-toolbar/issues/45

### CODE REVIEW

**Comment by ccatterina** [comment]
Hi @luisbocanegra , thank you so much for continuing to contribute to this project! I will look into this and the other PR in the next few days!

File: src/contents/ui/ScrollingText.qml [hunk 1]
```diff
@@ -23,7 +23,11 @@ Item {
     property int speed: 5;
     readonly property int duration: (25 * (11 - speed) + 25)* textAndSpacing.length;
 
+    property bool scrollingEnabled: true
+    property bool scrollResetOnPause: false
+
     readonly property bool pauseScrolling: {
+        if (!scrollingEnabled) return true
```
Comment (ccatterina) [inline src/contents/ui/ScrollingText.qml:30]: setting `paused` to True when the animation is not running cause this warning:
```sh
QML NumberAnimation: setPaused() cannot be used when animation isn't running
```

I think it's better to set animation `running` to False when is `scrollingEnabled` is False, something like:
```qml
running: root.overflow && root.scrollingEnabled
```

File: src/contents/ui/ScrollingText.qml [hunk 2]
```diff
@@ -84,6 +88,9 @@ Item {
             }
             onToChanged: () => reset()
             onDurationChanged: () =>  reset()
+            onPausedChanged: (paused) => {
+                if (paused && (scrollResetOnPause || !scrollingEnabled)) reset()
```
Comment (ccatterina) [inline src/contents/ui/ScrollingText.qml:92]:
```suggestion
                if (paused && scrollResetOnPause) reset()
```

I think that this `scrollingEnabled` check is no longer needed if you apply the suggestion in the other comment.

**Review by ccatterina — CHANGES_REQUESTED** [review]

File: src/contents/ui/ScrollingText.qml (hunk 1 above)
Comment (luisbocanegra) [reply src/contents/ui/ScrollingText.qml]: Yeah, before this parch I see that warning too when changing songs, turns out the paused property needs to check if it's running, and also the reset function. With that there is no warnings.

```qml
paused: root.pauseScrolling && running
```

```qml
function reset() {
      label.x = 0;
      if (running) {
          restart()
      }
      if (running && root.pauseScrolling) {
          pause()
      }
  }
```

File: src/contents/ui/ScrollingText.qml (hunk 2 above)
Comment (luisbocanegra) [reply src/contents/ui/ScrollingText.qml]: Done, also fixed a infinite loop caused by calling `reset()`, changed it to set `label.x = 0`

```qml
if (paused && scrollResetOnPause) label.x = 0
```

**Review by ccatterina — APPROVED** [review]

**Pull request merged**